TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -584,9 +584,16 @@ public static DataDefinition CheckVariable(Node node, StorageArea storageArea, b
                 if (!isReadStorageArea && node.SymbolTable.CurrentScope == SymbolTable.Scope.Function)
                 {
                     var paramDesc = (dataDefinitionPath?.CurrentDataDefinition ?? dataDefinitionFound) as ParameterDescription;
+                    //Check if we're dealing with an input parameter
                     if (paramDesc?.PassingType == ParameterDescription.PassingTypes.Input)
                     {
-                        DiagnosticUtils.AddError(node, "Input variable '" + paramDesc.Name + "' is modified by an instruction", area.SymbolReference);
+                        var specialRegister = storageArea as StorageAreaPropertySpecialRegister;
+                        //Unless this is a format 5 set statement, we have an error. So we're checking we're not in the following format  :
+                        //set (address of)? identifier(pointer) TO (address of)? identifier | NULL
+                        if (specialRegister?.SpecialRegisterName.TokenType != TokenType.ADDRESS || !specialRegister.IsWrittenTo)
```

Comment:
Is use of property `IsWrittenTo` required ? You already have `isReadStorageArea` in the method, is it the same ?